deploy/cloudformation/elastic-agent-ec2-cnvm.yml

```diff
@@ -142,14 +142,25 @@ Resources:
       UserData: !Base64
         Fn::Sub: |
           #!/bin/bash -x
+          signal() {
+            sudo yum install -y aws-cfn-bootstrap || { sudo apt-get update && sudo apt-get -y install python3-pip && sudo pip3 install https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-py3-latest.tar.gz; }
+            cfn-signal --exit-code "$1" --stack ${AWS::StackName} --resource ElasticAgentEc2Instance --region ${AWS::Region}
+          }
+          ## version_gte expects 2 version strings and returns code 0 if the first one is greater than or equal to the second.
+          version_gte() {
+            printf '%s\n%s\n' "$2" "$1" | sort --check=quiet --version-sort
+          }
+          AGENT_MIN_VERSION="$(echo 'CFT_VERSION' | sed --regexp-extended  's/^[a-zA-Z-]+-(.*)-[[:digit:]]{4}(-[[:digit:]]{2}){5}.yml/\1/g')"
```

Comment:
(Optional) It might be more readable to do the version checking with an inline python script

Reply:
Sure, I guess you mean to use something like the [semver](https://python-semver.readthedocs.io/en/latest/usage/compare-versions.html) package to write something like that `semver.compare("1.0.0", "2.0.0")` ? 

That would be more readable. 

Can we safely assume that the semver would be installed in the default vm? If not what could be the approach?

Is there any native way to do it in python? (I am not super familiar with python 😅)

Reply:
semver is not part of the standard library, so I don't think we should rely on it.

Something like this could be a starting point:
```shell
python -c '
import sys
from packaging.version import Version
a = sys.argv[1]
b = sys.argv[2]
print(Version(a) < Version(b))
' 1.0 2.0
```

Reply:
Nvm, `packaging` is also non-standard